### ISSUE TITLE
BUG: Dancing Mad doesn't fallback to SPC audio when no PCM is present

### ISSUE THREAD

**Issue by edale2** [Insidious611/DancingMadFF6#31]
So I wanted to check the timing of the SPC looping on Dancing Mad, so deleted ff3-82, ff3-101, ff3-102, and ff3-103.

Loaded up the final fight aaaaand.... silence. Only getting game SFX, no music at all.

---

Got a bit further in the fight. Track 4 (the short kefka final form intro) plays, and track 5 plays after that as well (so everything from track 82). Music also stops as kefka dies for some odd reason.

I'm guessing this has something to do with the way you coded things to rename the PCM audio to tracks 101-103.

**Comment by Dizzy611** [comment]
Ahh yeah, it's likely that it's being fed the invalid track numbers back to the SPC routines.

**Comment by Dizzy611** [comment]
Try the IPS from here.
https://github.com/Insidious611/DancingMadFF6/blob/spc-3b-fix/patch/ff3msu.ips

The branch name is a reference to the hex track number for the SPC dancing mad, 3b. :P This *should* be a fix, it fixes it for at least parts 1 and 2 from my testing (I didn't test the battle all the way to part 3 yet).
However, if one part is missing it will never play the MSU-1 for the subsequent tracks, even if they're there. Also, if you only have the first part but not the second, it might not play the correct part of the SPC for the second. Same if you have the first and second parts but not the third.

Basically, it's all of dancing mad 1-3 or none of them, all other behavior is undefined (and there's really no way for me to fix that).

There's a summary of what I did in the commit log, but basically I had it return the SPC track number (hex 3b/dec 59) to the SPC routine when it defaults to it, if it's trying to play one of the MSU-1-only Dancing Mad tracks (hex 65 66 67/dec 101 102 103) and can't find them. Before, it was returning 101/102/103 to these routines and the SPC was ignoring the invalid music track.

**Comment by edale2** [comment]
1 quick runthrough of the fight shows it working on all parts.  I'll do some more thorough testing after work tomorrow (well... later today, consider it tomorrow because i'm sleeping betwen then and now, lol) @

**Comment by edale2** [comment]
With only track 101 present: PSM for first part, silence until final form.

With only track 102 present; SPC for part 1, SPC + PSM for part 2, SPC for part 3.

With only track 103 present: SPC for parts 1 and 2, SPC + PSM for part 3.

**Comment by Dizzy611** [comment]
Yeah, that seems similar to what I expected. Unfortunately there's not a lot I can do to make it failthrough cleaner if you have one or two parts out of 3, not without further modifying the SPC code for the final battle and potentially introducing even more bugs.

**Comment by Dizzy611** [comment]
Going to merge the spc-3b-fix branch into the master branch and consider this bug closed for now, unless I find a better way to do this.

I'll let people know in the known bugs section of the readme that 101-103 are a package deal and to remove all of them if they remove any of them.
